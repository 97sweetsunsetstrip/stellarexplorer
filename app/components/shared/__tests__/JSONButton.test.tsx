import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import JSONButton from '../JSONButton'

describe('JSONButton', () => {
  it('renders button with given url', () => {
    const url = 'https://somebackend.xyz/resource/12345'

    render(<JSONButton url={url} />)

    const button = screen.getByText('JSON')
    expect(button).toBeInTheDocument()

    // snapshot test
    const tree = renderer.create(<JSONButton url={url} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
