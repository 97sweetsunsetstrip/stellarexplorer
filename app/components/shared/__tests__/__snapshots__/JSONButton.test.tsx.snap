// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONButton renders button with given url 1`] = `
<span>
  <a
    className="backend-resource-badge-button"
    onClick={[Function]}
  >
    JSON
  </a>
</span>
`;
